packages/psss-server/src/connections/ApiConnection.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { fetchWithTimeout, checkStatusAndParseResponse, stringifyQuery } from '@tupaia/utils';
+import { FetchConfig, QueryParameters, RequestBody } from '../types';
+
+interface AuthHandler {
+  getAuthHeader: () => Promise<string>;
+}
+
+export class ApiConnection {
```

Comment:
This class is awesome!! It's so good that we may find ourselves moving it in `utils` (`utils-ts`?) at some stage (not neccessary now)